contracts/state/StateV2.sol

```diff
@@ -69,7 +69,7 @@ contract StateV2 is Ownable2StepUpgradeable, IState {
     ) public initializer {
         verifier = verifierContractAddr;
         _defaultIdType = defaultIdType;
-        _transitionEnabledForOnchainIdentity = true;
+        _defaultIdTypeInitialized = true;
```

Comment:
Just call the `setDefaultIdType()` here

Reply:
We can't because it will fail. Msg.sender is not owner here.

Reply:
True. But the DRY is still violated. Better move the same code to internal method or modifier

Reply:
We can have `internal` setter, but still have to provide `external` method to be able to upgrade contract / change `_defaultIdType` value outside of contract. This `external` will call `internal` setter. As for me, better to have one `exteranal` method and some logic in initialize (even if it's duplication) rather that have 2 setters (internal&external).
@AndriianChestnykh what do you think ?

Reply:
Better to have a logic in one place. And two auth .... **wrappers**!!!.
Yes, I don't agree with your definition of setters. These are NOT setters (setter should be only one and that's the point) but just two wrappers, each of each implements different auth logic: one of them - `initializer`, which is called by ProxyAdmin contract and doing some init checks, another - onlyOwner modifier, checking for the `msg.sender == owner`). 
You may add 10 more auth wrappers in the future, which will be external methods and that will be fine in therms of design.
What tells you that design is good? The answer is - MAINTEINABILITY. Now both auth wrappers and the setter can evolve independently.

Reply:
Agreed, create a new PR with `internal` setter & 1 `external` wrapper (with onlyOwner modifier) - https://github.com/iden3/contracts/pull/156/files